File: patient/lib/models/transaction.dart
import 'package:flutter/material.dart';

import 'details.dart';

class Transaction {
  final String sender;
  final String receiver;
  final List<Details> details;

  final DateTime timestamp;

  Transaction({
    @required this.sender,
    @required this.receiver,
    @required this.details,
    @required this.timestamp,
  });
}

File: blockchain_backend/transaction.py
from collections import OrderedDict


class Transaction():
    def __init__(self, sender, recepient, signature, details,timestamp):
        self.sender = sender
        self.receiver = recepient
        self.details = details
        self.timestamp=timestamp
        self.signature = signature

    # create ordered dict to help in generating guess in valid proof
    def to_ordered_dict(self):
        return OrderedDict([
            ('sender', self.sender),
            ('receiver', self.receiver),
            ('details', self.details)
        ])

    def __repr__(self):
        return str(self.__dict__)

